ddtrace/contrib/kafka/patch.py

```diff
@@ -185,12 +198,28 @@ def traced_poll(func, instance, args, kwargs):
     if not pin or not pin.enabled():
         return func(*args, **kwargs)
 
-    with pin.tracer.trace(
-        schematize_messaging_operation(kafkax.CONSUME, provider="kafka", direction=SpanDirection.PROCESSING),
+    # we must get start time now since execute before starting a span in order to get distributed context
+    # if it exists
+    start_ns = time_ns()
+    # wrap in a try catch and raise exception after span is started
+    err = None
+    try:
+        message = func(*args, **kwargs)
+    except Exception as e:
+        err = e
+    ctx = None
+    if message and config.kafka.distributed_tracing_enabled and message.headers():
```

Comment:
Let's add in the "empty poll" toggle here as well

Reply:
Done